### PR TITLE
contract blocked list(cause smb)

### PR DESCRIPTION
**Pull request by RevelationOfTuring**
- Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/okex/okexchain/blob/develop/CONTRIBUTING.md#pr-targeting))

- [ ] Linked to github-issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote tests
- [ ] Updated relevant documentation (`docs/`)
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Reviewed `Files changed` in the github PR explorer

______

For Admin Use:
- Added appropriate labels to PR (ex. wip, ready-for-review, docs)
- Reviewers Assigned
- Squashed all commits, uses message "Merge PR #XYZ: [title]" ([coding standards](https://github.com/okex/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: x/evm/genesis.go [hunk 1]
```diff
@@ -164,5 +167,6 @@ func ExportGenesis(ctx sdk.Context, k Keeper, ak types.AccountKeeper) GenesisSta
 		ChainConfig:                 config,
 		Params:                      k.GetParams(ctx),
 		ContractDeploymentWhitelist: csdb.GetContractDeploymentWhitelist(),
+		ContractBlockedList:         csdb.GetContractBlockedList(),
```
Comment (xiangjianmeng) [inline x/evm/genesis.go:170]: It is better to implement  `GetContractDeploymentWhitelist` or `GetContractBlockedList` for keeper not csdb, I think that csdb is for interacting with evm.

File: x/evm/keeper/proposal.go [hunk 2]
```diff
@@ -72,11 +77,34 @@ func (k Keeper) CheckMsgManageContractDeploymentWhitelistProposal(ctx sdk.Contex
...
 }
+
+// CheckMsgManageContractBlockedListProposal checks msg manage contract blocked list proposal
+func (k Keeper) CheckMsgManageContractBlockedListProposal(ctx sdk.Context,
+	proposal types.ManageContractBlockedListProposal) sdk.Error {
+	csdb := types.CreateEmptyCommitStateDB(k.GeneratePureCSDBParams(), ctx)
+	if proposal.IsAdded {
+		// add contract addr into blocked list
+		// 1. check the existence
+		if csdb.IsContractInBlockedList(proposal.ContractAddr) {
```
Comment (xiangjianmeng) [inline x/evm/keeper/proposal.go]: use keeper not csdb

File: x/evm/types/genesis.go [hunk 3]
```diff
@@ -85,7 +86,7 @@ func DefaultGenesisState() GenesisState {
 	return GenesisState{
 		Accounts:                    []GenesisAccount{},
 		TxsLogs:                     []TransactionLogs{},
-		ContractDeploymentWhitelist: ContractDeploymentWhitelist{},
+		ContractDeploymentWhitelist: AddressList{},
```
Comment (xiangjianmeng) [inline x/evm/types/genesis.go:89]: lack the  initialization of `ContractBlockedList `

File: x/evm/types/state_transition.go [hunk 4]
```diff
@@ -113,12 +113,22 @@ func (st StateTransition) newEVM(
 // TransitionDb will transition the state by applying the current transaction and
 // returning the evm execution result.
 // NOTE: State transition checks are run during AnteHandler execution.
-func (st StateTransition) TransitionDb(ctx sdk.Context, config ChainConfig) (*ExecutionResult, *ResultData, error) {
+func (st StateTransition) TransitionDb(ctx sdk.Context, config ChainConfig) (exeRes *ExecutionResult, resData *ResultData, err error) {
+	defer func() {
+		if e := recover(); e != nil {
+			// if the msg recovered can be asserted into type 'common.Address', it must be captured by the panics of blocked
+			// contract calling
+			if blockedContractAddr, ok := e.(common.Address); ok {
```
Comment (KamiD) [inline x/evm/types/state_transition.go:121]: e.(common.Address) will failed if there has other panic which was not caused by blockedContract, need add type assert

**Review by KamiD — APPROVED** [review]

File: x/evm/keeper/proposal.go (hunk 2 above)
Comment (RevelationOfTuring) [reply x/evm/keeper/proposal.go]: It will cause package importing circulation when trying to invoke keeper in csdb.

**Review by xiangjianmeng — APPROVED** [review]

**Pull request merged**